src/connection.ts

```diff
@@ -1865,6 +1865,11 @@ class Connection extends EventEmitter {
   }
 
   beginTransaction(callback: (err: Error | null | undefined, transactionDescriptor?: Buffer) => void, name = '', isolationLevel = this.config.options.isolationLevel) {
+    const isoValues = Object.values(ISOLATION_LEVEL);
```

Comment:
How do you feel about extracting this check to the `Transaction` class, as a static method? There's multiple methods that take isolation level strings, and centralising the validation seems useful to me. Something along the lines of `Transaction.validateIsolationLevel`.

Additionally, how do you feel about changing the way the check is performed? `Object.values` always allocates a new `Array` object that holds the same numbers (`0`, `1`, `2`, `3`, `4`, `5`), and calling `.includes` will always iterate over these values to check if the given `isolationLevel` can be found. This can be simplified by doing a check that verifies `isolationLevel >= 0 && isolationLevel <= 5`, or even a simple `switch` statement like:

```js
switch (isolationLevel) {
  case ISOLATION_LEVEL.NO_CHANGE:
  case ISOLATION_LEVEL.READ_UNCOMMITTED:
  case ISOLATION_LEVEL.READ_COMMITTED:
  case ISOLATION_LEVEL.REPEATABLE_READ:
  case ISOLATION_LEVEL.SERIALIZABLE:
  case ISOLATION_LEVEL.SNAPSHOT:
    break;

  default:
    throw new Error(`Error: invalid isolationLevel ${isolationLevel}`);
}
```